api/client/cp.go

```diff
@@ -210,10 +212,28 @@ func (cli *DockerCli) copyFromContainer(srcContainer, srcPath, dstPath string) (
 		IsDir:  stat.Mode.IsDir(),
 	}
 
+	var content io.ReadCloser
+	if cli.isTerminalOut {
+		sf := streamformatter.NewStreamFormatter()
+		progressReader := progressreader.New(progressreader.Config{
+			In:             response.body,
+			Out:            cli.out,
+			Formatter:      sf,
+			Size:           stat.Size,
```

Comment:
If copy a dir, the `stat.Size` is wrong, it's not the totally size of the dir.
Maybe you can use https://github.com/docker/docker/blob/master/pkg/directory/directory_unix.go#L12 to get the total size of a directory if `stat.Mode.IsDir()` is `true`